### PR TITLE
Feature/asuprun/failover

### PR DESCRIPTION
**Pull request by aleksey-suprun**

### CODE REVIEW

File: test-utils/src/main/java/com/basho/riak/test/rule/cluster/RiakClusterRule.java [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+            ContainerCreation containerCreation = docker.createContainer(containerConfig, containerConfig.hostname());
+            docker.startContainer(containerCreation.id());
+
+            // wait until Riak process will be ready
+            dockerExec(containerCreation.id(), new String[]{"riak-admin", "wait-for-service", "riak_kv"});
+
+            NetworkSettings networkSettings = docker.inspectContainer(containerCreation.id()).networkSettings();
+            ipMap.put(containerCreation.id(), networkSettings.ipAddress());
+
+            if (StringUtils.isNotBlank(joinToId)) {
```
Comment (srgg) [inline test-utils/src/main/java/com/basho/riak/test/rule/cluster/RiakClusterRule.java]: is it a normal case when joinToId is blank? If not, we need to handle then

File: test-utils/src/main/java/com/basho/riak/test/rule/RiakClusterRule.java [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+            public void evaluate() throws Throwable {
+                try {
+                    CountDownLatch clusterStartLatch = new CountDownLatch(nodes);
+                    IntStream.range(0, nodes).forEach(i -> new Thread(() -> {
+                        startNodeContainer(clusterName, clusterName + i, imageName);
+                        clusterStartLatch.countDown();
+                    }).start());
+
+                    if (!clusterStartLatch.await(timeout, timeUnit)) {
+                        throw new IllegalStateException("The timeout period elapsed prior to Riak cluster start completion");
```
Comment (srgg) [inline test-utils/src/main/java/com/basho/riak/test/rule/RiakClusterRule.java]: Why don't us interrupt all the spanned threads?

Comment (aleksey-suprun) [inline test-utils/src/main/java/com/basho/riak/test/rule/RiakClusterRule.java]: Done

File: connector/src/test/scala/com/basho/riak/spark/rdd/failover/FailoverTest.scala [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  override def destroySparkContext(): Unit = {
+    Option(stubNodes).foreach(_.foreach(_.stop()))
+    super.destroySparkContext()
+  }
+}
+
+object FailoverTest {
+  final val NODES_AMOUNT = 3
+  final val STUBS_AMOUNT = 1
+  final val RIAK_IMAGE = "basho/riak-ts:1.3.0"
```
Comment (mdigan) [inline connector/src/test/scala/com/basho/riak/spark/rdd/failover/FailoverTest.scala]: Can we externalize this so we can update the docker image easily?

Comment (srgg) [inline connector/src/test/scala/com/basho/riak/spark/rdd/failover/FailoverTest.scala]: We may use ENV variable for this

Comment (aleksey-suprun) [inline connector/src/test/scala/com/basho/riak/spark/rdd/failover/FailoverTest.scala]: I think ENV variable is a good idea. I'll spend some time and add such functionality ASAP.

Comment (aleksey-suprun) [inline connector/src/test/scala/com/basho/riak/spark/rdd/failover/FailoverTest.scala]: "com.basho.riak.test.cluster.image-name" ENV variable introduced

**Pull request merged**